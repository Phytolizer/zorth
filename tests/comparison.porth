include "std.zorth"

// equals
69 420 = print
69 69  = print

"------------------------------\n" write

// greater
420 69 > print
69 420 > print

"------------------------------\n" write

// less
420 69 < print
69 420 < print

"------------------------------\n" write

// greater or equal
420 69 >= print
69 420 >= print
69 69 >= print

"------------------------------\n" write

// less or equal
420 69 <= print
69 420 <= print
69 69 <= print

"------------------------------\n" write

// not equal
69 69 != print
420 69 != print
