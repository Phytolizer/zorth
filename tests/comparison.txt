0
1
1111111111
1
0
1111111111
0
1
