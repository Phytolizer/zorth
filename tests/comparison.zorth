// equals
69 420 = print
69 69  = print

"------------------------------\n" 1 1 syscall3 drop

// greater
420 69 > print
69 420 > print

"------------------------------\n" 1 1 syscall3 drop

// less
420 69 < print
69 420 < print

"------------------------------\n" 1 1 syscall3 drop

// greater or equal
420 69 >= print
69 420 >= print
69 69 >= print

"------------------------------\n" 1 1 syscall3 drop

// less or equal
420 69 <= print
69 420 <= print
69 69 <= print

"------------------------------\n" 1 1 syscall3 drop

// not equal
69 69 != print
420 69 != print
