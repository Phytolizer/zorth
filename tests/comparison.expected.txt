0
1
111111111
1
0
111111111
0
1
